### PR TITLE
WRR-3892: Updated node modules` major versions

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] I have run automated CLI testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated node modules` major versions

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated node modules` major versions:
@stylistic/eslint-plugin-js        ^1.7.0  →  ^2.8.0
@typescript-eslint/eslint-plugin  ^7.7.1  →  ^8.6.0
@typescript-eslint/parser         ^7.7.1  →  ^8.6.0

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Updated Node engine version to `^18.18.0 || ^20.9.0 || >=21.1.0`

Change logs:
@stylistic/eslint-plugin-js
v2.0.0
https://github.com/eslint-stylistic/eslint-stylistic/releases/tag/v2.0.0
requires Node ^18.18.0 || ^20.9.0 || >=21.1.0

@typescript-eslint/eslint-plugin and @typescript-eslint/parser
v8.0.0
https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.0.0
parser: always enable comment, loc, range, tokens 
eslint-plugin: [prefer-nullish-coalescing] change ignoreConditionalTests default to true 
eslint-plugin: deprecate no-loss-of-precision extension rule 
eslint-plugin: remove formatting/layout rules 
eslint-plugin: [no-unused-vars] align catch behavior to ESLint 9 
eslint-plugin: split no-empty-object-type out from ban-types and no-empty-interfaces 
eslint-plugin: apply initial config changes for v8 
eslint-plugin: replace ban-types with no-restricted-types, no-unsafe-function-type, no-wrapper-object-types 
eslint-plugin: remove no-useless-template-literals 
eslint-plugin: [no-floating-promises] disable checkThenables by default for v8 

### Links
[//]: # (Related issues, references)
WRR-3892

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ The following is a curated list of changes in the Enact eslint config:
 ## [unreleased]
 
 * Replaced deprecated rule `no-new-object` with `no-object-constructor`.
+* Updated the minimum version of Node to `^18.18.0 || ^20.9.0 || >=21.1.0`.
+* Updated `@stylistic/eslint-plugin-js` to `^2.8.0`, `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` to `^8.6.0`.
+* Updated the `lockfileVersion` of npm-shrinkwrap file to v3.
```
Comment (hong6316) [inline CHANGELOG.md:10]: I think we don't need it.

Comment (hong6316) [inline CHANGELOG.md:10]:
```suggestion
```

Comment (hong6316) [reply CHANGELOG.md:10]: I think we don't need it

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**